backend/src/akvo/lumen/lib/update.clj

```diff
@@ -126,55 +126,61 @@
-               imported-table-name (util/gen-table-name "imported")]
-           (postgres/create-dataset-table conn table-name importer-columns)
-           (doseq [record (map (comp postgres/coerce-to-sql import/extract-first-and-merge) (p/records importer))]
-             (jdbc/insert! conn table-name record))
-           (db.job-execution/clone-data-table conn {:from-table table-name
-                                                    :to-table   imported-table-name}
-                                              {}
-                                              {:transaction? false})
-           (let [coerce-column-fn (fn [{:keys [title id type key multipleId multipleType groupName groupId] :as column}]
-                                    (cond-> {"type" type
-                                             "title" title
-                                             "columnName" id
-                                             "groupName" groupName
-                                             "groupId" groupId
-                                             "sort" nil
-                                             "direction" nil
-                                             "hidden" false}
-                                      key           (assoc "key" (boolean key))
-                                      multipleType (assoc "multipleType" multipleType)
-                                      multipleId   (assoc "multipleId" multipleId)))]
-             {:success? true
-              :table-name table-name :imported-table-name imported-table-name
-              :importer-columns (mapv coerce-column-fn importer-columns) :imported-dataset-columns imported-dataset-columns
-              :latest-dataset-version latest-dataset-version})))))))
+              columns-used (columns-used-in-txs
+                            (import/importer-type (get data-source-spec "source"))
+                            initial-dataset-version
+                            latest-dataset-version)
+              imported-dataset-columns-checked (reduce (fn [c co]
+                                                         (if (contains? columns-used (get co "columnName"))
+                                                           (conj c co)
+                                                           c)) [] imported-dataset-columns)]
+          (if-let [compatible-errors (compatible-columns-error? imported-dataset-columns-checked
+                                                                (if (get environment "optionColumnType")
+                                                                  (map #(update % :type (fn [t]
+                                                                                          (if (= "option" t) "text" t)))
+                                                                       importer-columns)
+                                                                  importer-columns))]
```

Comment:
and here, that we depending feature flag "optionColumnType" we avoid mismatch column types due to `OPTION` type